internals/secrethub/rm_test.go

```diff
@@ -0,0 +1,265 @@
+			},
+			argPath: "namespace/repo/dir",
+			promptOut: "[WARNING] This action cannot be undone. This will permanently remove the namespace/repo/dir directory and all the directories and secrets it contains. " +
+				"Please type in the name of the directory to confirm: ",
+			in:  "namespace/repo/dir",
+			out: "Removal complete! The directory namespace/repo/dir has been permanently removed.\n",
+		},
+		"fail non recursive dir": {
+			cmd: RmCommand{
+				force:     true,
+				recursive: false,
+				path:      "namespace/repo/dir",
+			},
+			argPath: "namespace/repo/dir",
+			err:     ErrCannotRemoveDir,
+		},
+		"fail remove root dir": {
+			cmd: RmCommand{
+				force:     true,
+				recursive: false,
+				path:      "namespace/repo",
+			},
+			argPath: "namespace/repo",
+			err:     ErrCannotRemoveRootDir,
+		},
+		"fail get tree error dir": {
+			cmd: RmCommand{
+				force:     true,
+				recursive: true,
+				path:      "namespace/repo/dir",
+			},
+			argPath:    "namespace/repo/dir",
+			getTreeErr: testErr,
+			err:        testErr,
+		},
+		"fail abort dir": {
+			cmd: RmCommand{
+				path:      "namespace/repo/dir",
+				recursive: true,
+				force:     false,
+			},
+			argPath: "namespace/repo/dir",
+			promptOut: "[WARNING] This action cannot be undone. This will permanently remove the namespace/repo/dir directory and all the directories and secrets it contains. " +
+				"Please type in the name of the directory to confirm: ",
+			in:  "namespace/repo/directory",
+			out: "Name does not match. Aborting.\n",
+		},
+		"fail client error dir": {
+			cmd: RmCommand{
+				path: "namespace/repo/dir",
+			},
+			err:          testErr,
+			newClientErr: testErr,
+		},
+		"fail deletion error dir": {
+			cmd: RmCommand{
+				path:      "namespace/repo/dir",
+				force:     true,
+				recursive: true,
+			},
+			argPath:      "namespace/repo/dir",
+			err:          testErr,
+			deleteDirErr: testErr,
+		},
+		"fail prompt error dir": {
+			cmd: RmCommand{
+				path:      "namespace/repo/dir",
+				recursive: true,
+			},
+			promptErr: testErr,
+			err:       testErr,
+		},
+		"success force secret version": {
+			cmd: RmCommand{
+				path:  "namespace/repo/secret:latest",
+				force: true,
+			},
+			argPath: "namespace/repo/secret:latest",
+			out:     "Removal complete! The secret version namespace/repo/secret:latest has been permanently removed.\n",
+		},
+		"success non force secret version": {
+			cmd: RmCommand{
+				force: false,
+				path:  "namespace/repo/secret:latest",
+			},
+			argPath: "namespace/repo/secret:latest",
+			promptOut: "[WARNING] This action cannot be undone. This will permanently remove the namespace/repo/secret:latest secret version. " +
+				"Please type in the name of the secret and the version (<name>:<version>) to confirm: ",
+			in:  "namespace/repo/secret:latest",
+			out: "Removal complete! The secret version namespace/repo/secret:latest has been permanently removed.\n",
```

Comment:
We could also make a constant for the warning and the removal complete messages.